src/env.js

```diff
@@ -28,24 +28,62 @@ const defaultEnvSource = (() => {
 
 })();
 
+//-----------------------------------------------------------------------------
+// Errors
+//-----------------------------------------------------------------------------
+
 /**
- * Throws an error saying that the key was not found.
- * @param {PropertyKey} key The key to report as missing.
- * @returns {void}
- * @throws {Error} Always. 
+ * The error thrown when a required key is not found.
  */
-function keyNotFound(key) {
-    throw new Error(`Required environment variable '${String(key)}' not found.`);
+export class EnvKeyNotFoundError extends Error {
+
+    /**
+     * Creates a new instance.
+     * @param {PropertyKey} key The key that wasn't found. 
```

Comment:
```suggestion
     * @param {string} key The key that wasn't found. 
```

This should be `string`.  Even though `PropertyKey` is allowed on any object, the keys of `NodeJS.ProcessEnv` are always `string`.

Ref: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/process.d.ts#L112

Reply:
Gotcha, makes sense. 👍